src/Interface/HasEmptySet.agda

```diff
@@ -0,0 +1,11 @@
+{-# OPTIONS --safe #-}
+
+open import Axiom.Set using (Theory)
+
+module Interface.HasEmptySet {ℓ} (th : Theory {ℓ}) where
+
+open Theory th using () renaming (Set to ℙ_)
+
+record HasEmptySet (A : Set ℓ) : Set ℓ where
+  field
+    ∅ : ℙ A
```

Comment:
```suggestion
record HasEmptySet (A : Set ℓ) : Set ℓ where
  field
    ∅ : A

instance
  _ : {A : Set} → HasEmptySet (ℙ A)
  _ = record { ∅ = ∅ˢ }

  _ : {A B : Set} → HasEmptySet (Map A B)
  _ = record { ∅ = ∅ᵐ }
```

Reply:
Okay, thanks, I'll try it.

Reply:
@WhatisRT This commit creates a lot of cyclic module dependencies in the set theory files, but since those are `.agda` files and won't show up in the pdf, I can choose to not import `HasEmptySet` there and simply add superscripts to the emptyset symbol in those modules.  (HasEmptySet now needs to import both `Axiom/Set` *and* `Axiom/Map`, so these cannot use `HasEmptySet`.)

That's fine, but now the problem is that `Set` has type `Type l -> Type l` whereas `Map` has type `Type -> Type -> Type`, so the levels don't line up in your unified module `HasEmptySet {l}`.  I'll keep working on it and try to resolve this.

Reply:
I think I figured it out.  Will push the fix soon.